prov/efa/src/rdm/efa_rdm_msg.c

```diff
@@ -854,13 +854,25 @@ struct efa_rdm_ope *efa_rdm_msg_alloc_rxe_for_tagrtm(struct efa_rdm_ep *ep,
 	size_t data_size;
 	int ret;
 	int pkt_type;
+	struct efa_rdm_base_hdr *base_hdr;
+	uint64_t cq_data, cq_flags;
 
 	peer_srx = util_get_peer_srx(ep->peer_srx_ep);
 	data_size = efa_rdm_pke_get_rtm_msg_length(*pkt_entry_ptr);
 
+	base_hdr = efa_rdm_pke_get_base_hdr(*pkt_entry_ptr);
+	if (base_hdr->flags & EFA_RDM_REQ_OPT_CQ_DATA_HDR) {
+		cq_flags = FI_REMOTE_CQ_DATA;
```

Comment:
Is `FI_REMOTE_CQ_DATA` enough here? Do we need to |= `ofi_rx_cq_flags(ofi_op_tagged)` ?

Reply:
So far, the peer srx AP flags are only defined for FI_REMOTE_CQ_DATA so I think for now, yes. The peer utility saves the FI_TAGGED | FI_RECV flags in since it knows it is a tagged receive. We should think about more concretely defining the ownership of the flags but that would require extending the flags to the get_msg API as well

Reply:
>The peer utility saves the FI_TAGGED | FI_RECV flags in since it knows it is a tagged receive.

Which peer utility saves those flags?

Reply:
sorry I meant the util srx here in these changes does that

Reply:
I missed this change 

```
util_entry = util_init_unexp(srx_ctx, addr, size, tag, cq_data,
				     FI_TAGGED | FI_RECV | flags);
```
Should be good, thanks!